packages/presentation/lib/screens/login/login_screen.dart

```diff
@@ -74,122 +74,142 @@ class _LoginScreenState extends BlocScreenState<LoginScreen, LoginBloc> {
-                            style: const TextStyle(
-                              color: AppColors.transparentWhite50,
-                            ),
-                            decoration: InputDecoration(
-                              disabledBorder: InputBorder.none,
-                              filled: true,
-                              fillColor:
-                                  LoginScreenColors.inputFieldBackgroundColor,
-                              prefixIcon: Image.asset(
-                                AssetsImagesPaths.passwordIconPath,
+                              controller: bloc.loginController,
+                              keyboardType: TextInputType.emailAddress,
+                              style: const TextStyle(
+                                color: AppColors.transparentWhite50,
+                              ),
+                              decoration: InputDecoration(
+                                disabledBorder: InputBorder.none,
+                                filled: true,
+                                fillColor:
+                                    LoginScreenColors.inputFieldBackgroundColor,
+                                prefixIcon: Image.asset(
+                                  AssetsImagesPaths.usernameIconPath,
+                                ),
                               ),
                             ),
-                          ),
-                          Padding(
-                            padding: const EdgeInsets.symmetric(
-                              vertical: AppSizes.size32,
+                            const SizedBox(height: AppSizes.size16),
+                            Text(
+                              appLocalizations.passwordInputLabel.toUpperCase(),
+                              style: LoginScreenStyles.inputLabelsStyle,
                             ),
-                            child: Center(
-                              child: Text(
-                                appLocalizations.forgotPasswordButtonLabel,
-                                style:
-                                    LoginScreenStyles.forgotPasswordLabelStyle,
+                            const SizedBox(height: AppSizes.size12),
+                            TextFormField(
+                              autovalidateMode: AutovalidateMode.disabled,
+                              validator: (_) => bloc.loginViewMapper
+                                  .stateToPasswordErrorMessage(
+                                bloc.state,
+                                authFailurePasswordMessage:
+                                    appLocalizations.authFailurePasswordMessage,
+                                requiredPasswordMessage:
+                                    appLocalizations.requiredPasswordMessage,
+                                invalidPasswordMessage:
+                                    appLocalizations.requiredPasswordMessage,
                               ),
```

Comment:
I think u can provide context as parameter to method of viewmapper to avoid passing each message(what if u'll have 100 messages?)

Reply:
Fixed.